arduino-core/src/processing/app/Serial.java

```diff
@@ -295,10 +267,13 @@ public void setRTS(boolean state) {
    * before they are handed as Strings to {@Link #message(char[], int)}.
    */
   public synchronized void resetDecoding(Charset charset) {
-    bytesToStrings = charset.newDecoder()
-                      .onMalformedInput(CodingErrorAction.REPLACE)
-                      .onUnmappableCharacter(CodingErrorAction.REPLACE)
-                      .replaceWith("\u2e2e");
+    try {
+      decoderInRaw = new PipedOutputStream();
+      decoderOutputUTF8 = new InputStreamReader(new PipedInputStream(decoderInRaw), charset);
```

Comment:
Any idea how `InputStreamReader` handles malformed or unmappable characters? I could not find anything in the docs quickly. If it's different than specified in the old code, that might be a behaviour change that would be good to document.

Reply:
I tried with an invalid UTF-8 sequence it is converted into the REPLACEMENT CHARACTER � (U+FFFD or `EF BF BD`), I've addded a test for it.

Reply:
Ok, so that slightly changes behaviour (it was U+2E2E, the upside down question mark). It seems that the replacement char is rendered as � (at least in my browser, haven't checked what the serial terminal does with it), so that's probably fine.